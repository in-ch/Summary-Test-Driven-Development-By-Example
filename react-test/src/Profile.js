import React from 'react';

const Profile = ({ username, name }) => {
  return (
    <div>
      <b>{username}</b>&nbsp;
      <span>({name})</span>
    </div>
  );
};

export default Profile;