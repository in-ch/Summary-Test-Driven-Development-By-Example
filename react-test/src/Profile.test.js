import React from 'react';
import { render } from '@testing-library/react';
import Profile from './Profile';

describe('<Profile />', () => {
  it('matches snapshot', () => {
    const utils = render(<Profile username="in-ch" name="인쵸리" />);
    expect(utils.container).toMatchSnapshot();
  });
  it('shows the props correctly', () => {
    const utils = render(<Profile username="in-ch" name="인쵸리" />);
    utils.getByText('in-ch');
    utils.getByText('(인쵸리)'); 
    utils.getByText(/인/);
  });
});