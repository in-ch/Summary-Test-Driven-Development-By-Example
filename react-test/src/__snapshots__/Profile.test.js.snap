// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      in-ch
    </b>
     
    <span>
      (
      인쵸리
      )
    </span>
  </div>
</div>
`;
